### DISCUSSION TITLE
[Question] Notes on using Contact Sensor

### DISCUSSION THREAD

**Discussion by shaoxiang** [isaac-sim/IsaacLab#594] [Q&A]
### Question about Contact Sensor 

I added a contact sensor according to the example steps, 
[https://isaac-sim.github.io/IsaacLab/source/tutorials/04_sensors/add_sensors_on_robot.html](url)
but when contact or collision occurs, the values in data.net_forces_w or data.net_forces_w_history are both 0, and the Contact Report in the prims physics in the scene has also been added. I also tried to add a callback according to the ContactReportDemo example in omni. physx. demos \ omni \ physxdemos, like: 
`self._contact_report_sub = get_physx_simulation_interface().subscribe_contact_report_events(self._on_contact_report_event)`
but unfortunately, I couldn't listen for any contact events. I think I must have gotten something wrong. I also added collision attributes to objects. Can you provide more detailed tutorials, usage instructions, or precautions about contact sensors?

**Comment by shaoxiang** [comment]
I solved this problem by adding this sentence, Now this part of the code looks like this：
```python
# add contact sensor
self._contact_sensor = ContactSensor(self.cfg.contact_sensor)
self.scene.sensors["contact_sensor"] = self._contact_sensor
```
I'm confused why I need to pass ``self._contact_sensor`` to ``self.scene.sensors ["contact_sensor"]``, This step solved my problem!

**Comment by Mayankm96** [comment]
Potentially, you are not calling the method to update the sensor data. This is implicitly handled by the `scene.update()` which iterates over all the entities and calls their methods to update the buffers.
